### PR TITLE
Replace quickstart with new getting started guide

### PR DESCRIPTION
**Pull request by rohanjr**
This moves the new guide into the Getting Started section, following Installation. I moved the IOU Quickstart to the Java Bindings section. I did not change its content besides the title.

I also updated the many references to the Quickstart to the new location, but I did not attempt to update them to point to the new guide when it makes sense. This is really just a minimal change to get the new guide in the right spot.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by nemanja-da** [comment]
Thnx for the great work @rohanjr! Any chance you could squeeze in this PR a small change for the GSG? The footnote in the app architecture section for the React Ledger library is broken (nothing to do with this PR though). 

On another note: why isn't the testing your app section part of the GSG?

File: docs/redirects.map [hunk 1]
```diff
@@ -10,7 +10,7 @@ daml/anti-patterns.html
 daml/models-versus-application.html
 daml/stdlib/base.html -> daml/reference/base.html
 concepts.html -> /concepts/ledger-model/index.html
-getting-started.html -> /getting-started/quickstart.html
+getting-started.html -> /app-dev/bindings-java/quickstart.html
```
Comment (bame-da) [inline docs/redirects.map:13]: I think we should point this to the new GSG. We are allowed to change the content of pages, we just don't want dead links.

File: docs/redirects.map [hunk 2]
```diff
@@ -52,7 +52,7 @@ packages/navigator-docs/database.html -> /tools/navigator/database.html
 packages/navigator-docs/backend-licenses.html -> /
 packages/navigator-docs/console.html -> /tools/navigator/console.html
 packages/navigator-docs/frontend-licenses.html -> /
-packages/quickstart/index.html -> /getting-started/quickstart.html
+packages/quickstart/index.html -> /app-dev/bindings-java/quickstart.html
```
Comment (bame-da) [inline docs/redirects.map:55]: Same here

File: docs/redirects.map (hunk 1 above)
Comment (rohanjr) [reply docs/redirects.map]: I don't actually know what this redirects map is, but I made the change.

File: docs/redirects.map (hunk 2 above)
Comment (rohanjr) [reply docs/redirects.map]: done

**Comment by rohanjr** [comment]
@nemanja-da I fixed that link

**Pull request merged**